### PR TITLE
feat(Drawer): add Drawer component

### PR DESCRIPTION
**Pull request by rogerioshieh**
Drawer component that opens up via an animation:

https://user-images.githubusercontent.com/40327446/208700326-e3ca2e55-20b2-4632-89f4-b263c5db21ab.mov

Component's API:
image

### CODE REVIEW

File: src/Drawer/index.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  };
+
+  /* eslint-disable jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events */
+  const navigationContainerJSX = (
+    <div
+      ref={navRef}
+      onClick={handleShimClick}
+      style={depthStyle}
+      className={cc([
+        `drawer navigation ${position}`,
```
Comment (akdetrick) [inline src/Drawer/index.js:95]: This is definitely the right approach to composing the classes here - for the sake of consistency, I'd like to change some of the class names to match our [naming conventions](https://github.com/narmi/design_system/blob/main/contributing/naming-conventions.md#css)

For example:
```diff
-.drawer.bottom
+.drawer--bottom
```

```diff
-.drawer.in
+.drawer--transitioning
```

This convention helps the component stay consistent with the naming in the rest of the project and also flattens the CSS specificity of each variant state to `10`

**Review by akdetrick — CHANGES_REQUESTED** [review]
Looking good!
Left a couple of comments for minor changes

File: src/Drawer/index.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  const isHorizontal = position === "bottom" || position === "top";
+  // The depth is how far the drawer opens out, but the CSS prop depends
+  // on whether the Drawer is vertical or not
+  const depthStyle = isHorizontal ? { height: depth } : { width: depth };
+  useLockBodyScroll(isOpen);
+
+  // Navigation buttons on vertical drawers are the opposite compared to
+  // horizontal ones due to the order of navigation buttons being the opposite
+  //(rows are reversed in parent divs for horizontal drawers)
+  const prevOrNext = (opposite = false) => {
```
[the remark this answers is not in the document]
Comment (akdetrick) [reply src/Drawer/index.js:42]: On closer inspection, it looks like we already use CSS to change the order of buttons.
Do we actually need `prevOrNext`, or should the callbacks be invoked directly?

[the remark this answers is not in the document]
Comment (rogerioshieh) [reply src/Drawer/index.js]: I am not exactly sure why I had done it that way at first, but you're totally right--I simplified the logic now to call them directly

**Review by akdetrick — APPROVED** [review]
🚢

**Pull request merged**